circuit_knitting/cutting/cut_wire_to_move.py

```diff
@@ -82,3 +86,63 @@ def _circuit_structure_mapping(
         new_circuit.add_register(creg)
 
     return new_circuit, mapping
+
+
+def expand_observables(
+    observables: PauliList,
+    original_circuit: QuantumCircuit,
+    final_circuit: QuantumCircuit,
+    /,
+) -> PauliList:
+    """Expand observable(s) according to the qubit mapping between ``original_circuit`` and ``final_circuit``.
+
+    The qubits on ``final_circuit`` must be a superset of those on
+    ``original_circuit``.
+
+    Given a :class:`.PauliList` of observables, this function returns new
+    observables with identity operators placed on the qubits that did not
+    exist in ``original_circuit``.  This way, observables on
+    ``original_circuit`` can be mapped to appropriate observables on
+    ``final_circuit``.
+
+    This function is designed to be used after calling ``final_circuit =
+    transform_cuts_to_moves(original_circuit)`` (see
+    :func:`.transform_cuts_to_moves`).
+
+    This function requires ``observables.num_qubits ==
+    original_circuit.num_qubits`` and returns new observables such that
+    ``retval.num_qubits == final_circuit.num_qubits``.
+
+    Args:
+        observables: Observables corresponding to ``original_circuit``
+        original_circuit: Original circuit
+        final_circuit: Final circuit, whose qubits the original ``observables`` should be expanded to.
+
+    Returns:
+        New observables, appropriate for the ``final_circuit``.
+
+    Raises:
+        ValueError: ``observables`` and ``original_circuit`` have different number of qubits.
+        ValueError: Qubit from ``original_circuit`` cannot be found in ``final_circuit``.
+    """
+    if observables.num_qubits != original_circuit.num_qubits:
+        raise ValueError(
+            "The `observables` and `original_circuit` must have the same number "
+            f"of qubits. ({observables.num_qubits} != {original_circuit.num_qubits})"
+        )
+    mapping: list[int] = []
+    for i, qubit in enumerate(original_circuit.qubits):
+        try:
+            idx = final_circuit.find_bit(qubit)[0]
+        except CircuitError as ex:
+            raise ValueError(
+                f"The {i}-th qubit of the `original_circuit` cannot be found "
```

Comment:
So the original and final circuits need to be comprised of the same `Qubit` instances, is that right?

If that's the case, we could add a sentence here telling users explicitly to build their two circuit using the same qubit instances. Or maybe this sentence is clear enough? What do you think?

Reply:
The change in 82f1e6a3e25f5cc0aa1e52f82355979e862cbac7 is subtle, but I think it helps to clarify this.  Since this function is meant to be used with `cut_wires` and `cut_wires` already maintains this condition, I don't think we need to belabor it IMO.